server/session.go

```diff
@@ -110,7 +118,20 @@ func (p *Plugin) removeUserSession(userID, channelID string) (channelState, chan
 
 		currState = *state
 		return state, nil
-	})
+	}
+
+	var err error
+	maxTries := 5
+	for i := 0; i < maxTries; i++ {
+		err = p.kvSetAtomicChannelState(channelID, setChannelState)
```

Comment:
I wonder if we should start to use the `mattermost-plugin-api`? It has `SetAtomicWithRetries` and a few other convenience functions that are well tested.  (I can't remember if I've mentioned that already, apologies if I have)

Reply:
Maybe. I am thinking we could probably use that in place of the `kvSetAtomic` util but this is different and would require a customized version because we need the attempts to continue under certain conditions that are implementation specific (in this case the session being there as we expect it to be).